core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -5364,6 +5364,24 @@ public static int localTime(DataContext root) {
     return timestampToTime(localTimestamp(root));
   }
 
+  /** SQL {@code SYSTIMESTAMP} function. */
+  @NonDeterministic
+  public static long sysTimestamp(DataContext root) {
+    return DataContext.Variable.SYS_TIMESTAMP.get(root);
+  }
+
+  /** SQL {@code SYSDATE} function. */
+  @NonDeterministic
+  public static int sysDate(DataContext root) {
+    final long timestamp = sysTimestamp(root);
+    int date = timestampToDate(timestamp);
+    final int time = timestampToTime(timestamp);
```

Comment:
why would this happen?
it it for dates before 1970?

Reply:
Yes, when the date is before `1970-1-1`, for example: `1969-12-31 23:59:59`, timestamp will return `-1000`, and the date(days since epoch) calculated by `timestampToDate(-1000)` is 0, so we need to additionally judge the result of `timestampToTime(-1000)`. If its value is less than 0, we need to reduce date by 1 to ensure the accuracy of date.

Reply:
this would be a nice comment to add to this code